commons-executors/src/test/java/com/palantir/common/concurrent/PTExecutorsTest.java

```diff
@@ -67,6 +74,31 @@ public void testExecutorName_customThreadFactory_fallback() {
         assertThat(PTExecutors.getExecutorName(factory)).isEqualTo("PTExecutor");
     }
 
+    @Test
+    public void testExecutorThreadLocalState_cachedPoolNospan() {
+        Tracer.initTraceWithSpan(Observability.SAMPLE, "RandomID", "outerThreadTest", SpanType.LOCAL);
+
+        String outerTraceId = Tracer.getTraceId();
+        AtomicReference<String> innerSpanId = new AtomicReference<>("");
+        AtomicReference<String> innerSpanParentId = new AtomicReference<>("");
+        AtomicReference<String> innerTraceId = new AtomicReference<>("");
+
+        withExecutor(PTExecutors::newCachedThreadPoolWithoutSpan, executor -> {
+            OpenSpan innerSpan = Tracer.startSpan("innerThreadTest");
+
+            innerTraceId.set(Tracer.getTraceId());
+            innerSpanId.set(innerSpan.getSpanId());
+            innerSpanParentId.set(innerSpan.getParentSpanId().get());
+
+            Tracer.fastCompleteSpan();
+        });
+        Span outerSpan = Tracer.completeSpan().get();
+
```

Comment:
But you are not submitting anything to the executor without span?
I think the test might be a bit too smart of slightly hard to read for me. Would you mind adding a few comments?

Reply:
I'll re-add comments.  I will also admit that I might be doing something wrong with the concurrency that makes the test do something dumb.

Reply:
I think I see what I was doing wrong, pushing new test.

Reply:
yeah this looks correct now

Reply:
ok this looks correct now!